tests/test_ble_connection_discovery_client_targets.py

```
@@ -0,0 +1,1045 @@
+    orchestrator.client_manager = SimpleNamespace(connect_client=_connect_client)
+    orchestrator._client_manager_connect_client(DummyClient(), timeout=1.0)
+    assert connect_calls == [{"timeout": 1.0}, {}]
+
+    iface.close()
+
+
+def test_connection_orchestrator_direct_and_retry_exception_paths(
+    monkeypatch: pytest.MonkeyPatch,
+) -> None:
+    """Direct/retry connect helpers should close clients on exceptional paths."""
+    iface, orchestrator = _make_orchestrator(monkeypatch)
```

Ruff (0.15.6)

[warning] Unpacked variable `iface` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)